### ISSUE TITLE
Pluggable SCM duplication causes config to silently become invalid, for some plugins

### ISSUE THREAD

**Issue by arvindsv** [gocd/gocd#7976]
##### Issue Type

- Bug Report

###### Basic environment details

* Go Version: 20.3.0 (11467-1fb7c3cc3a71abb71e827f5bbdc2cb139edcd660).

##### Steps to Reproduce

1. Get the [GitPath material plugin](https://github.com/TWChennai/gocd-git-path-material-plugin/releases/tag/1.2.3) and the [GitHub PR plugin](https://github.com/ashwanthkumar/gocd-build-github-pull-requests/releases/tag/v1.4.0-RC2).
2. Use this  config:

    ```xml
    <scms>
      <scm id="5ba163bf-f69c-42cd-a21c-cdc2ef09290c" name="m1">
        <pluginConfiguration id="GitPathMaterial" version="1" />
        <configuration>
          <property>
            <key>url</key>
            <value>/tmp/test.git</value>
          </property>
          <property>
            <key>username</key>
            <value>a</value>
          </property>
          <property>
            <key>password</key>
            <encryptedValue />
          </property>
          <property>
            <key>path</key>
            <value />
          </property>
          <property>
            <key>branch</key>
            <value />
          </property>
        </configuration>
      </scm>
      <scm id="8028ed2f-7a57-4419-9555-d52f3cf3f1a2" name="pr1" autoUpdate="false">
        <pluginConfiguration id="github.pr" version="1" />
        <configuration>
          <property>
            <key>url</key>
            <value>/tmp/test.git</value>
          </property>
          <property>
            <key>username</key>
            <value />
          </property>
          <property>
            <key>password</key>
            <encryptedValue />
          </property>
          <property>
            <key>defaultBranch</key>
            <value />
          </property>
          <property>
            <key>shallowClone</key>
            <value />
          </property>
        </configuration>
      </scm>
    </scms>
    ```

3. Try to clone each of these SCMs through the pluggable SCMs UI. Make no changes. Just provide a name.

##### Expected Results

They should both fail (duplicate SCMs).

##### Actual Results

The (old) GitPathMaterial plugin succeeds and causes the config to be corrupted. You can verify it by going to the config XML and trying to re-save it or by trying to restart the server (won't come back up).

The GitHub PR plugin refuses to save (correctly) and shows that the SCM is a duplicate.

Don't know if it's a bug with the plugin and its validation. But, the fact that the config is _silently_ made invalid is a little worrying.

**Comment by kritika-singh3** [comment]
Issue cause similar to what is described over here - https://github.com/gocd/gocd/issues/7974#issuecomment-610650663

**Comment by GaneshSPatil** [comment]
Fixed as part of #7982

Verified on `GoCD Version: 20.3.0 (11467-1fb7c3cc3a71abb71e827f5bbdc2cb139edcd660).`

The save fails with following error:
<img width="1101" alt="Screen Shot 2020-04-10 at 11 41 31 AM" src="https://user-images.githubusercontent.com/15275847/78967531-60b70b00-7b20-11ea-96d3-b2c6b9d32e31.png">
